src/param.rs

```diff
@@ -328,8 +364,50 @@ impl AudioParamProcessor {
             //
             //        self.insert_event(event);
             //            ^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
-            //
-            // if no event in the timeline and event_type is `LinearRampToValueAtTime`
+
+            // handle CancelScheduledValues events
+            // cf. https://www.w3.org/TR/webaudio/#dom-audioparam-cancelscheduledvalues
+            if event.event_type == AutomationType::CancelScheduledValues {
+                let some_next_event = self.events.peek();
+
+                match some_next_event {
+                    None => (),
+                    Some(next_event) => {
+                        match next_event.event_type {
+                            AutomationType::LinearRampToValueAtTime
+                            | AutomationType::ExponentialRampToValueAtTime => {
+                                // @note - Firefox and Chrome behave differently
+                                // on this: Firefox actually restore intrisic_value
+                                // from the value at the beginning of the vent, while
+                                // Chrome just keeps the current intrisic_value
+                                // The spec is not very clear there, but Firefox
+                                // seems to be the more compliant:
+                                // "Any active automations whose automation event
+                                // time is less than cancelTime are also cancelled,
+                                // and such cancellations may cause discontinuities
+                                // because the original value (**from before such
+                                // automation**) is restored immediately."
+                                if next_event.time >= event.time {
+                                    let last_event = self.last_event.unwrap();
```

Comment:
Is it possible that `self.last_event = None` at this point?

Reply:
Good point, but no after verification that's not possible, because a linear or exponential ramps are always preceded by another event (even a set_value_at_time inserted implicitly), so if the ramp is the next event that means that at least one event has already been processed.

I added a comment in the code, as well as tests to prevent possible regressions. If you feel it's more safe, I can also add a condition, it wouldn't hurt neither

Reply:
A test is fine!